planner/cmd/planner/main.go

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2025 Canonical Ltd.
+ * See LICENSE file for licensing details.
+ *
+ * Package main starts the Planner API service.
+ *
+ * This service connects to the database, exposes a REST API for managing GitHub runner
+ * flavors, and provides observability features via Prometheus metrics and OpenTelemetry tracing.
+ */
+
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"os"
+
+	"github.com/canonical/github-runner-operators/internal/auth"
+	"github.com/canonical/github-runner-operators/internal/database"
+	"github.com/canonical/github-runner-operators/planner"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+const dbURI = "PLANNER_DATABASE_URI"
```

Comment:
I think we intend to use the 12-factor application which has the following environment variables for database:
https://documentation.ubuntu.com/charmcraft/latest/howto/manage-web-app-charms/integrate-web-app-charm/#integrate-with-a-database

They are also common database environment variables, so we could use it instead of `PLANNER_DATABASE_URI` :)